### PR TITLE
Fix panic and tidy up exporter logic

### PR DESCRIPTION
**Pull request by rossf7**
While testing #71 I hit a bug when the API token has expired.

This causes a panic which makes the exporter crash and need a restart. It should just log the error.

```
2023/07/12 18:18:39 failed to get carbon intensity &fmt.wrapError{msg:"401 Unauthorized - {\"error\":\"Invalid auth-token\"}: received non-200 status", err:(*errors.errorString)(0x14000180ce0)}
panic: runtime error: index out of range [0] with length 0

goroutine 34 [running]:
github.com/thegreenwebfoundation/grid-intensity-go/cmd.(*Exporter).Collect(0x14000078280, 0x140000b6758?)
	/Users/ross/src/rossf7/grid-intensity-go/cmd/exporter.go:191 +0x2a8
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	/Users/ross/.gvm/pkgsets/go1.20.5/global/pkg/mod/github.com/prometheus/client_golang@v1.15.1/prometheus/registry.go:455 +0xd8
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/Users/ross/.gvm/pkgsets/go1.20.5/global/pkg/mod/github.com/prometheus/client_golang@v1.15.1/prometheus/registry.go:547 +0x9f4
exit status 2
```

### CODE REVIEW

File: cmd/exporter.go [hunk 1]
```diff
@@ -150,54 +150,27 @@ func NewExporter(config ExporterConfig) (*Exporter, error) {
...
-		result, err := e.client.GetCarbonIntensity(ctx, e.location)
+	for _, data := range result {
+		desc, err := getMetricDesc(data)
 		if err != nil {
-			log.Printf("failed to get carbon intensity %#v", err)
+			log.Printf("failed to get metric description %#v", err)
+			continue
 		}
 
-		averageIntensity := result[0]
```
Comment (rossf7) [inline cmd/exporter.go:191]: This is the cause of the panic as result will be nil if there is an error

File: cmd/exporter.go [hunk 2]
```diff
@@ -150,54 +150,27 @@ func NewExporter(config ExporterConfig) (*Exporter, error) {
...
-					data.Provider,
-					e.region,
-					data.Units,
-				)
-			}
-		}
-	} else {
-		result, err := e.client.GetCarbonIntensity(ctx, e.location)
+	for _, data := range result {
+		desc, err := getMetricDesc(data)
```
Comment (rossf7) [inline cmd/exporter.go:159]: Looping through the results prevents the panic. 

I also took the chance to clean up the code a bit.

**Comment by rossf7** [comment]
@mrchrisadams Can you also take a look at this PR when you have time. Thx!

**Comment by mrchrisadams** [comment]
Hi Ross! This looks great - thanks for the PR. I'm merging in.

**Pull request merged**